pkg/testsuite/benchmarks_test.go

```diff
@@ -0,0 +1,578 @@
+			block = block2.ProtocolBlock()
+			modelBlock = lo.PanicOnErr(model.BlockFromBlock(block))
+			b.StartTimer()
+			// time from issuance of the block to when it is scheduled
+			node.Protocol.IssueBlock(modelBlock)
+			<-blockchan
+			b.StopTimer()
+			ts.Shutdown()
+		}
+	}
+	// get the ns/op of processing the block
+	nsPerBlock := float64(testing.Benchmark(fn).NsPerOp())
+	fmt.Printf("One native token in one native token output: %f ns/op\n", nsPerBlock)
+	// get the regressors
+	regressors := getBlockWorkScoreRegressors(block)
+	printRegressors(regressors)
+
+	return nsPerBlock, regressors
+}
+
+func allotments(t *testing.T, numAllotments int) (float64, []float64) {
+	blockchan := make(chan *blocks.Block, 1)
+	var block *iotago.Block
+
+	// basic block with one input and one output
+	fn := func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			b.StopTimer()
+			ts := NewTestSuite(t)
+			// create genesis accounts to allot to.
+			for i := 0; i < numAllotments; i++ {
+				ts.AddGenesisAccount(snapshotcreator.AccountDetails{
+					Address:              nil,
+					Amount:               mock.MinIssuerAccountAmount(ts.API.ProtocolParameters()),
+					Mana:                 0,
+					IssuerKey:            tpkg.RandBlockIssuerKey(),
+					ExpirySlot:           iotago.MaxSlotIndex,
+					BlockIssuanceCredits: iotago.BlockIssuanceCredits(123),
+				})
+			}
+			node := ts.AddValidatorNode("node1")
+			ts.AddDefaultWallet(node)
+			ts.Run(true)
+			var accountIDs []iotago.AccountID
+			for i := 0; i < numAllotments; i++ {
+				accountOutput := ts.AccountOutput(fmt.Sprintf("Genesis:%d", i+1)).Output().(*iotago.AccountOutput)
+				accountIDs = append(accountIDs, accountOutput.AccountID)
+			}
+			tx1 := ts.DefaultWallet().AllotManaFromInput(
+				"tx1",
+				"Genesis:0",
+				accountIDs...,
+			)
+			// default block issuer issues a block containing the transaction in slot 1.
+			genesisCommitment := iotago.NewEmptyCommitment(ts.API)
+			genesisCommitment.ReferenceManaCost = ts.API.ProtocolParameters().CongestionControlParameters().MinReferenceManaCost
+			block1 := ts.IssueBasicBlockWithOptions("block1", ts.DefaultWallet(), tx1, mock.WithSlotCommitment(genesisCommitment))
+			block = block1.ProtocolBlock()
+			modelBlock := lo.PanicOnErr(model.BlockFromBlock(block))
+			node.Protocol.Events.Engine.Scheduler.BlockScheduled.Hook(func(block *blocks.Block) {
+				blockchan <- block
+			})
+			b.StartTimer()
+			// time from issuance of the block to when it is scheduled
+			node.Protocol.IssueBlock(modelBlock)
+			<-blockchan
+			b.StopTimer()
+			ts.Shutdown()
+		}
+	}
+	// get the ns/op of processing the block
+	nsPerBlock := float64(testing.Benchmark(fn).NsPerOp())
+	fmt.Printf("One input, %d allotments: %f ns/op\n", numAllotments, nsPerBlock)
+	// get the regressors
+	regressors := getBlockWorkScoreRegressors(block)
+	printRegressors(regressors)
+
+	return nsPerBlock, regressors
```

Comment:
A lot of the code repeats itself for all the different scenarios. Maybe we can abstract that into one function, that does all the "stop timer, init test suite, call func to prepare genesis, hook to the block event, call func for the specific test case, start timer and issue block, etc..."